### ISSUE TITLE
The statement is always true in 'rai::active_transactions::announce_votes' ？

### ISSUE THREAD

**Issue by huye0728** [nanocurrency/raiblocks#724]
size_t announcements(0)
{
    ...
        for (; i != n && **announcements < announcements_per_interval**; ++i)
       {
            ...
          // lost announcements ++ in loop, announcements < announcements_per_interval alway true
       }
}
// lost announcements ++ in loop???

**Comment by PlasmaPower** [comment]
Oh that's an interesting bug. To sum it up `announcements` is never incremented.

**Comment by huye0728** [comment]
if this is a bug, i think 'announcements_per_interval = 32' is too small, because 'announce_interval_ms = (rai::rai_network == rai::rai_networks::rai_test_network) ? 10 : 16000',  only 32 blocks republished every 16 seconds?

**Comment by PlasmaPower** [comment]
Yeah it's definitely too small, so it's a good thing that it isn't currently working. There are some better WIP rate limiting solutions.
